### PR TITLE
Extend collections cleanup task to include playlists too

### PR DESCRIPTION
**Pull request by Shadowghost**
Playlists have the same issue as collections with items not being removed from them when they are removed.

**Changes**
* Extend collection cleanup task to clean up playlists too.

**Issues**
Partially addresses #10216

### CODE REVIEW

**Comment by crobibero** [comment]
ci is failing with 
```
/home/vsts/work/1/s/Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs(112,90): error SA1127: Generic type constraints should be on their own line 
```

File: Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+                CleanupLinkedChildren(collection, cancellationToken);
+                progress.Report(50D / collections.Length * (index + 1));
+            }
+        }
+
+        var playlistsFolder = _playlistManager.GetPlaylistsFolder();
+        if (playlistsFolder is null)
+        {
+            _logger.LogDebug("There is no collection folder to be found");
```
Comment (crobibero) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs:95]:
```suggestion
            _logger.LogDebug("There is no playlists folder to be found");
```

File: Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            return;
+        }
+
+        var playlists = playlistsFolder.Children.OfType<Playlist>().ToArray();
+        _logger.LogDebug("Found {PlaylistLength} boxsets", playlists.Length);
```
Comment (crobibero) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs:100]:
```suggestion
        _logger.LogDebug("Found {PlaylistLength} playlists", playlists.Length);
```

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        foreach (var linkedChild in folder.LinkedChildren)
+        {
+            if (!File.Exists(folder.Path))
+            {
+                _logger.LogInformation("Item in {FolderName} cannot be found at {ItemPath}", folder.Name, linkedChild.Path);
+                itemsToRemove.Add(linkedChild);
+            }
+        }
+
+        if (itemsToRemove.Count != 0)
```
Comment (Bond-009) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs:125]:
```suggestion
        List<LinkedChild> itemsToRemove = null;
        foreach (var linkedChild in folder.LinkedChildren)
        {
            if (!File.Exists(folder.Path))
            {
                _logger.LogInformation("Item in {FolderName} cannot be found at {ItemPath}", folder.Name, linkedChild.Path);
                (itemsToRemove ??= new List<LinkedChild>()).Add(linkedChild);
            }
        }

        if (itemsToRemove is not null)
```

File: Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+            _providerManager.QueueRefresh(
+                folder.Id,
+                new MetadataRefreshOptions(new DirectoryService(_fileSystem))
+                {
+                    ForceSave = true
+                },
+                RefreshPriority.High);
+
+            itemsToRemove.Clear();
```
Comment (Bond-009) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs:139]:
```suggestion
```

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+            CleanupLinkedChildren(playlist, cancellationToken);
+            progress.Report(50D / playlists.Length * (index + 1));
+        }
+    }
+
+    private void CleanupLinkedChildren<T>(T folder, CancellationToken cancellationToken)
+        where T : Folder
+    {
+        List<LinkedChild> itemsToRemove = null;
```
Comment (Bond-009) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs:115]:
```suggestion
        List<LinkedChild>? itemsToRemove = null;
```
Should fix build error

File: Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            folder.UpdateToRepositoryAsync(ItemUpdateType.MetadataEdit, cancellationToken);
+
+            _providerManager.QueueRefresh(
+                folder.Id,
+                new MetadataRefreshOptions(new DirectoryService(_fileSystem))
+                {
+                    ForceSave = true
+                },
+                RefreshPriority.High);
+
```
Comment (Bond-009) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/CleanupCollectionAndPlaylistPathsTask.cs:138]:
```suggestion
```
Should be the last issue

**Review by JPVenson — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**